src/CPUSolver.cpp

```diff
@@ -550,11 +540,13 @@ double CPUSolver::computeResidual(residualType res_type) {
     FP_PRECISION inverse_k_eff = 1.0 / _k_eff;
     FP_PRECISION* nu_sigma_f;
     Material* material;
+    FP_PRECISION* sigma_s;
 
     for (int r=0; r < _num_FSRs; r++) {
```

Comment:
I remember Matt Everson running into some performance issues with the calculation for the scattering source taking a long time when many energy groups were used. While not directly related to this PR, seeing this for loop reminded me of that issue. Could you parallelize this loop to avoid any performance bottlenecks?